src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -293,5 +293,15 @@ private static bool IsDeconstructionCompatibleArgument(ExpressionSyntax expressi
 
             return false;
         }
+
+        internal static Location GetLocationForDiagnostics(this SyntaxNode node)
```

Comment:
Not sure this should be a general extension on `SyntaxNode` where the only use is in `ReportCantConvertLambdaReturn` and only to specifically handle lambdas.  
I think this should just be a private helper next to `ReportCantConvertLambdaReturn` .